src/domain/challenge/challenge/challenge.entity.ts

```diff
@@ -56,10 +52,21 @@ export class Challenge extends BaseEntity implements IChallenge, IGroupable {
   @JoinColumn()
   context?: Context;
 
+  @Field(() => Community, {
+    nullable: true,
+    description: 'The community for the challenge',
+  })
+  @OneToOne(
+    () => Community,
+    community => community.challenge,
+    { eager: false, cascade: true, onDelete: 'CASCADE' }
+  )
+  @JoinColumn()
+  community?: Community;
```

Comment:
onDelete: 'CASCADE' must be only on the one side of a relation, otherwise will lead to cyclic reference.

Reply:
Fixed in https://github.com/cherrytwist/server/pull/821/commits/689c369cafab7d4253aa0b358830932b593ed16c

Note that onDelete behavior needs a careful discussion - a lot of discussion on forums about this.